src/test/java/uk/gov/hmcts/reform/professionalapi/util/FeatureConditionEvaluationTest.java

```diff
@@ -0,0 +1,106 @@
+import uk.gov.hmcts.reform.professionalapi.controller.WelcomeController;
+import uk.gov.hmcts.reform.professionalapi.exception.ForbiddenException;
+import uk.gov.hmcts.reform.professionalapi.service.impl.FeatureToggleServiceImpl;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.professionalapi.util.FeatureConditionEvaluation.BEARER;
+import static uk.gov.hmcts.reform.professionalapi.util.FeatureConditionEvaluation.SERVICE_AUTHORIZATION;
+
+public class FeatureConditionEvaluationTest {
+
+    FeatureToggleServiceImpl featureToggleService = mock(FeatureToggleServiceImpl.class);
+    FeatureConditionEvaluation featureConditionEvaluation = new FeatureConditionEvaluation(featureToggleService);
+    HttpServletRequest httpRequest = mock(HttpServletRequest.class);
+    HttpServletResponse httpServletResponse = mock(HttpServletResponse.class);
+    HandlerMethod handlerMethod = mock(HandlerMethod.class);
+    Method method = mock(Method.class);
+
+    @Before
+    public void before() {
+        MockitoAnnotations.openMocks(this);
+        when(method.getName()).thenReturn("test");
+        doReturn(WelcomeController.class).when(method).getDeclaringClass();
+        when(handlerMethod.getMethod()).thenReturn(method);
+    }
+
+    @Test
+    public void testPreHandleValidFlag() throws Exception {
+        Map<String, String> launchDarklyMap = new HashMap<>();
+        launchDarklyMap.put("WelcomeController.test", "test-flag");
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(httpRequest));
+        when(featureToggleService.getLaunchDarklyMap()).thenReturn(launchDarklyMap);
+        String token = generateDummyS2SToken("rd_professional_api");
+        when(httpRequest.getHeader(SERVICE_AUTHORIZATION)).thenReturn(BEARER + token);
+        when(featureToggleService.isFlagEnabled(anyString(),anyString())).thenReturn(true);
+        assertTrue(featureConditionEvaluation.preHandle(httpRequest, httpServletResponse, handlerMethod));
+    }
+
+    @Test(expected = ForbiddenException.class)
+    public void testPreHandleInvalidFlag() throws Exception {
+        Map<String, String> launchDarklyMap = new HashMap<>();
+        launchDarklyMap.put("WelcomeController.test", "test-flag");
+        when(featureToggleService.getLaunchDarklyMap()).thenReturn(launchDarklyMap);
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(httpRequest));
+        String token = generateDummyS2SToken("rd_professional_api");
+        when(httpRequest.getHeader(SERVICE_AUTHORIZATION)).thenReturn(BEARER + token);
+        when(featureToggleService.isFlagEnabled(anyString(),anyString())).thenReturn(false);
```

Comment:
same as above

Reply:
why should we use verify? I did not see any point as it not adds any values. Assert checks actual results.

Reply:
As to verify that the `when()` has actually been used, in addition to this it was part of the QA Concerns document we received some months ago, that all uses of `when()` will require a `verify()`

Reply:
I don't think so. I think ideal case to use Verify in functional with sernity. & it is redundant to use Verify in Junit/Integration tests.And all above are junit/Integration tests.

Reply:
Either way, this is a QA Concern from HMCTS, we will need to add `verify()` for any `when()`...

Reply:
I think QA concerns should be in functional cases & not unit/integration. lets @mallikarjunputtana confirm this.

Reply:
for now done all verify :)